### ISSUE TITLE
[Core] Stop the DSA PR

### ISSUE THREAD

**Issue by mkubdev** [TheAlgorithms/Solidity#61]
Hello 👋🏻 

This repos is taking a wrong direction. Solidity language isn't suited for heavy mathematics computing.

I think we need to remove the list that looks like DSA, and restructure the repos to list Solidity use cases instead.

**Comment by iSumitBanik** [comment]
@mkubdev , Let me know what do you think about #58 . I just added the missing tests in this.
And, Let me also know if #60 is invalid considering the points you mentioned in this issue?

Thanks :)

**Comment by sreekar9601** [comment]
We should make sub-sections in a way such that it's more useful for implementation,
example: Algos commonly used in DeFi, NFTs, etc.

**Comment by Perelyn-sama** [comment]
@sreekar9601 great idea

@mkubdev I think continuing won't be a bad idea. 
Yeah, solidity is not the best language for DSA but when we attempt to write common algos in it, we learn more about the language.

Also, I think we have to give a thought to writing algos in Yul and Huff since they're more suited for mathematical operations.

@iSumitBanik and  @sreekar9601  thanks for your contributions, its nice to not be alone anymore.

**Comment by iSumitBanik** [comment]
> Also, I think we have to give a thought to writing algos in Yul and Huff since they're more suited for mathematical operations.

Agreed with this point. Writing in Yul/Huff would let us contribute some heavily gas optimized contracts ;)

**Comment by priyda** [comment]
> Hello 👋🏻
> 
> This repos is taking a wrong direction. Solidity language isn't suited for heavy mathematics computing.
> 
> I think we need to remove the list that looks like DSA, and restructure the repos to list Solidity use cases instead.

@mkubdev - I would be interested in contributing for non DSA, I am learning solidity by example. Should I add some simple contracts which I am trying for learning the language features?

**Comment by mkubdev** [comment]
> @mkubdev - I would be interested in contributing for non DSA, I am learning solidity by example. Should I add some simple contracts which I am trying for learning the language features?

How do you think packaging this examples?

**Comment by priyda** [comment]
> > @mkubdev - I would be interested in contributing for non DSA, I am learning solidity by example. Should I add some simple contracts which I am trying for learning the language features?
> 
> How do you think packaging this examples?

@mkubdev - I have raised a pull request, can you please review that once? If looks good. I have more to add for your review :)

**Comment by mkubdev** [comment]
> > > @mkubdev - I would be interested in contributing for non DSA, I am learning solidity by example. Should I add some simple contracts which I am trying for learning the language features?
> > 
> > 
> > How do you think packaging this examples?
> 
> @mkubdev - I have raised a pull request #77, can you please review that once? If looks good. I have more to add for your review :)

Sure, i'll review it as soon as i'm available. :D

**Comment by 0xSumitBanik** [comment]
> > > @mkubdev - I would be interested in contributing for non DSA, I am learning solidity by example. Should I add some simple contracts which I am trying for learning the language features?
> > 
> > How do you think packaging this examples?
> 
> @mkubdev - I have raised a pull request https://github.com/TheAlgorithms/Solidity/pull/77, can you please review that once? If looks good. I have more to add for your review :)

I added a few comments from my side.
